### PR TITLE
[BEAM-4199][SQL] Add support for DLQ to PubsubJsonTable

### PR DESCRIPTION
**Pull request by akedin**
Allows specifying DLQ where the messages would go when they cannot be parsed

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by akedin** [comment]
R: @kennknowles @apilloud

**Comment by akedin** [comment]
run java postcommit

**Comment by akedin** [comment]
green the first try

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubIOJsonTable.java [hunk 1]
```diff
@@ -123,12 +141,32 @@ public BeamIOType getSourceType() {
...
   @Override
   public PCollection<Row> buildIOReader(Pipeline pipeline) {
-    return
+    PCollectionTuple rowsWithDlq =
         PBegin
             .in(pipeline)
             .apply("readFromPubsub", readMessagesWithAttributes())
-            .apply("parseMessageToRow", PubsubMessageToRow.forSchema(getSchema()))
-            .setCoder(getSchema().getRowCoder());
+            .apply("parseMessageToRow", createParserParDo());
```
Comment (kennknowles) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubIOJsonTable.java]: A note for the future is that building the layers very carefully as composite PTansforms gives good rendering in UIs and places you can add things like DisplayData.

So you might have the nesting:

 - PubsubTable
   - PubsubIO
   - ParseToRow
      - JsonToRowWithDeadLetter
      - WriteDeadLetter

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java [hunk 2]
```diff
@@ -112,6 +112,7 @@ private void initializePubsub(Description description) throws IOException {
     //    integ-test-TestClassName-testMethodName-2018-12-11-23-32-333-<random-long>-result
     String resultTopicPathTmp =
         String.format(TOPIC_FORMAT, pipelineOptions.getProject(), createTopicName(description));
+    System.out.println("      >>> signal topic path:" + resultTopicPathTmp);
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java]: Let's not write to stdout.

**Comment by akedin** [comment]
run java postcommit

**Comment by akedin** [comment]
run java precommit

**Comment by akedin** [comment]
run java precommit

**Comment by akedin** [comment]
mine is green as well

**Pull request merged**